src/wallet/coinselection.h

```diff
@@ -247,6 +248,27 @@ struct Groups {
     std::vector<OutputGroup> mixed_group;
 };
 
+/** Stores several 'Groups' whose were mapped by output type. */
+struct OutputGroups
+{
+    // Map filters to output groups.
+    std::map<OutputType, Groups> groups_by_type;
+    // All groups, no filters
+    Groups all_groups;
+
+    enum class InsertGroupType {
+        ONLY_POSITIVE, // Insert only to `positive_group`
+        MIXED_GROUPS, // Insert only to `mixed_group`
+        BOTH // Insert to `positive_group` and `mixed_group`
+    };
+
+    // Depending on the 'insert_type', appends the output group into the 'by type' groups
+    // and/or the 'all groups' container.
```

Comment:
IIUC, for all possible `insert_types`, something is inserted both in the `by type` groups and `all groups` container. The `and/or` seems to apply for the `positive_group`/`mixed_group`? Suggestion:
```suggestion
    // Depending on the 'insert_type', appends the output group into the `positive_group`
    // and/or the 'mixed_group' for both the `by type` groups and the `all groups` container.
```

Reply:
Yeah, thanks!.
Aside from the method doc re-phrasing, have re-worked the area, so it's clear what the `push` method does on each scenario.